src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -324,6 +330,44 @@ private static IEnumerable<VisualStudioInstance> GetInstances(VisualStudioInstan
         }
 
 #if NET46
+        /// <summary>
+        /// Gets CoreXT instances.
+        /// </summary>
+        /// <param name="visualStudioInstances">Optional <see cref="IList{VisualStudioInstance} "/> containing instances.  If none are specified, they are discovered through the Visual Studio Setup API.
+        /// This is only here for unit tests</param>
+        /// <param name="directoryExists">An optional <see cref="Func{String,Boolean} "/> to use when checking directory existence.  Should only be set by unit tests.</param>
+        /// <returns>A <see cref="VisualStudioInstance" /> object representing the Visual Studio instance if one is found, otherwise null.</returns>
+        internal static VisualStudioInstance GetCoreXTInstance(IList<VisualStudioInstance> visualStudioInstances = null, Func<string, bool> directoryExists = null)
+        {
+            var toolset = Environment.GetEnvironmentVariable("MsBuildToolset")?.Trim();
+
+            if (string.IsNullOrWhiteSpace(toolset) || !int.TryParse(toolset, out int toolsetVersion) || toolsetVersion < 150)
+            {
+                // MsBuildToolset is not set or is less than 150
+                return null;
+            }
+
+            var msbuildPath = Environment.GetEnvironmentVariable($"MSBuildToolsPath_{toolset}");
+
+            if (string.IsNullOrWhiteSpace(msbuildPath) || !(directoryExists ?? Directory.Exists)(msbuildPath))
+            {
+                // A corresponding MSBuildToolsPath_XXX env var must be set and it must exist
+                return null;
+            }
+
+            if (!Version.TryParse(Environment.GetEnvironmentVariable("VisualStudioVersion"), out Version visualStudioVersion))
+            {
+                // VisualStudioVersion env var must be set to a valid version
+                return null;
+            }
+
+            // Find the newest installed Visual Studio instance that corresponds with the VisualStudioVersion set by CoreXT
+            VisualStudioInstance visualStudioInstance = (visualStudioInstances ?? VisualStudioLocationHelper.GetInstances()).Where(i => i.Version.Major == visualStudioVersion.Major).OrderByDescending(i => i.Version).FirstOrDefault();
```

Comment:
I'm confused by this. I would have expected that in a CoreXT environment we'd load only the MSBuild.CoreXT-package delivered MSBuild; this instead loads . . . a hybrid of the CoreXT targets and the installed-VS engine?

Reply:
As far as I can tell, the only property used when registering an instance is the MSBuildPath.  So this logic essentially lets callers know the path to Visual Studio in case they want to use it but the MSBuild path is the CoreXT one and that is the path used for loading assemblies.  Technically there is no Visual Studio path for the CoreXT environment so I thought this was a cool feature to associate the closest installed VS for the MSBuild.Corext package in use.

Reply:
This doesn't seem that useful to me. Can we not, and add it in later if we see a concrete use case?

Reply:
Without this, my primary use case I need it for has to query the instances separately.  This is very useful for me to replace all of this code:

https://github.com/microsoft/slngen/blob/master/src/Microsoft.VisualStudio.SlnGen/Program.Configure.cs#L26-L44

Of course this would just be a default, users could still manually find VS instances by calling `QueryVisualStudioInstances`.  But I feel that most callers would just want to know the best VS installation for the specified `MSBuild.Corext`.

Reply:
I'm going to stop using `Microsoft.Build.Locator` for my application since I need to locate MSBuild and Visual Studio.  I'll remove the code that tries to find a VS instance for CoreXT and just keep this PR about finding only MSBuild.